# Kendall Tau

In [ ]:
import scipy.stats as stats

P_10 = [.3566,.1818,.0584,.0519,.5479,.1883,.2974,.0039,.2403,.2390,.1390]
MRR = [.6103,.4339,.1003,.0854,.864,.2811,.4829,.0237,.4196,.3857,.2432]

tau, p_value = stats.kendalltau(P_10, MRR)

print("tau: " + str(tau))
print("P-Value: " + str(p_value))

tau: 0.8909090909090909
P-Value: 1.3728555395222063e-05


**Results**

File|P @ 10|MRR
----------|------------|---------
Prim_approach0|0.3566|0.6103
Prim_combined|0.1818|0.4339
Prim_dprl_A|0.0584|0.1003
Prim_dprl_P|0.0519|0.0854
Prim_linked|0.5479|0.864
Prim_math_auto|0.1883|0.2811
Prim_math_norerank|0.2974|0.4829
Prim_mirmu_bert|0.0039|0.0237
Prim_mirmu_ensemble|0.2403|0.4196
Prim_psu_A|0.2390|0.3857
Prim_psu_P|0.1390|0.2432

tau: 0.8909090909090909

P-Value: 1.3728555395222063e-05


**Discussion**

Precision and MRR are very similar, but slightly different. Where precision measures only at the cut, and doesn't care where the results were in the ranking, MRR takes into account where the relevant documents showed up in the ranking. For MRR, a number closer to 1 indicates that your relevant documents were at the top of the ranking, where if they were closer to 0, they would not be there at all, or at the bottom.

With how similar the statistics are, there should be a correlation between the two. This is proven with a very strong tau score of 0.89, which indicates a strong, positive correlation, and a p-value of almost 0, meaning the chances that the results are inaccurate are very low. 